lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -452,6 +461,45 @@ prop_listRegisteredPools DBLayer {..} entries =
             ]
         assert (pools == (poolId <$> reverse entries))
 
+prop_unfetchedPoolMetadataRefs
+    :: DBLayer IO
+    -> [PoolRegistrationCertificate]
+    -> Property
+prop_unfetchedPoolMetadataRefs DBLayer{..} entries =
+    monadicIO (setup >> propA >> propB)
+  where
+    setup = do
+        run . atomically $ cleanDB
+        let entries' = (zip [SlotId ep 0 | ep <- [0..]] entries)
+        run . atomically $ mapM_ (uncurry putPoolRegistration) entries'
+        monitor $ classify (length entries > 10) "10+ entries"
+        monitor $ classify (length entries > 50) "50+ entries"
+
+    propA = do
+        refs <- run . atomically $ unfetchedPoolMetadataRefs 10
+        monitor $ counterexample $ unlines
+            [ "Read from DB (" <> show (length refs) <> "): " <> show refs
+            ]
+        assertWith "no more than known entries"
+            (length refs <= length entries)
+        assertWith "all pool ids are indeed known"
+            (all (\(pid,_,_) -> pid `elem` (poolId <$> entries)) refs)
+        assertWith "no duplicate"
+            (L.nub refs == refs)
+
+    propB = do
+        refs <- run . atomically $ unfetchedPoolMetadataRefs 10
+        unless (null refs) $ do
+            let [(poolId, url, hash)] = take 1 refs
+            metadata <- pick $ genStakePoolMetadata url
+            run . atomically $ putPoolMetadata poolId hash metadata
+            refs' <- run . atomically $ unfetchedPoolMetadataRefs 10
+            monitor $ counterexample $ unlines
+                [ "Read from DB (" <> show (length refs') <> "): " <> show refs'
+                ]
+            assertWith "fetched metadata isn't listed anymore"
+                (hash `notElem` ((\(_,_,c) -> c) <$> refs'))
```

Comment:
This seems like an important property, but the names `prop_unfetchedPoolMetadataRefs` and `propB` makes it difficult to see.

Maybe just merging `propA` and `propB` would make it clearer.

Reply:
> Maybe just merging propA and propB would make it clearer.

There's nothing really to merge. In _theory_, we should really have two distinct properties but I was a bit lazy to rewrite the setup and function signature so I just added it quickly after. I could name `propA` and `propB` a bit better.